neo/Network/P2P/TaskManager.cs

```diff
@@ -219,24 +203,9 @@ private void RequestTasks(TaskSession session)
                     return;
                 }
             }
-            if ((!HasHeaderTask || globalTasks[HeaderTaskHash] < MaxConncurrentTasks) && Blockchain.Singleton.HeaderHeight < session.StartHeight)
-            {
-                session.Tasks[HeaderTaskHash] = DateTime.UtcNow;
-                IncrementGlobalTask(HeaderTaskHash);
-                session.RemoteNode.Tell(Message.Create(MessageCommand.GetHeaders, GetBlocksPayload.Create(Blockchain.Singleton.CurrentHeaderHash)));
-            }
-            else if (Blockchain.Singleton.Height < session.StartHeight)
+            if (Blockchain.Singleton.Height < session.StartHeight)
             {
                 UInt256 hash = Blockchain.Singleton.CurrentBlockHash;
-                for (uint i = Blockchain.Singleton.Height + 1; i <= Blockchain.Singleton.HeaderHeight; i++)
-                {
-                    hash = Blockchain.Singleton.GetBlockHash(i);
-                    if (!globalTasks.ContainsKey(hash))
-                    {
-                        hash = Blockchain.Singleton.GetBlockHash(i - 1);
-                        break;
-                    }
-                }
                 session.RemoteNode.Tell(Message.Create(MessageCommand.GetBlocks, GetBlocksPayload.Create(hash)));
```

Comment:
This will be answered with the hashes, maybe we should ask for the block

Reply:
This could be improved in p2p protocol.